test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/PushCommandTest.cs

```diff
@@ -245,6 +256,512 @@ public void PushCommand_Server_SkipDuplicate_IsSpecified_PushProceeds()
+                Util.CreateTestPackage(packageId, version, packageDirectory);
+                string nupkgFileName2 = Util.BuildPackageString(packageId, version, NuGetConstants.PackageExtension);
+                string snupkgFileName2 = Util.BuildPackageString(packageId, version, NuGetConstants.SnupkgExtension);
+                string snupkgFullPath2 = Path.Combine(packageDirectory, snupkgFileName2);
+                //Create another snupkg in test directory.
+                WriteSnupkgFile(snupkgFullPath2);
+
+                string wildcardPush = "*.nupkg";
+
+                using (var server = CreateAndStartMockV3Server(packageDirectory, out string sourceName))
+                {
+                    SetupMockServerAlwaysDuplicate(server);
+
+                    // Act
+
+                    //Since this is V3, this will trigger 2 pushes: one for nupkgs, and one for snupkgs.
+                    var result = CommandRunner.Run(
+                        nuget,
+                        packageDirectory,
+                        $"push {wildcardPush} -Source {sourceName} -Timeout 110 -SkipDuplicate",
+                        waitForExit: true,
+                        timeOutInMilliseconds: 120000); // 120 seconds
+
+                    // Assert
+
+                    //Ignoring filename in File Not Found error since the error should not appear in any case.
+                    string genericFileNotFoundError = string.Format(MESSAGE_FILE_DOES_NOT_EXIST, string.Empty);
+
+                    //Nupkg should be an ignored conflict, so its snupkg should push.
+                    //TODO: Once SkipDuplicate is passed-through to the inherit snupkg push, the following TODO's should be corrected.
+                    Assert.False(result.Success, "Expected to fail to push a snupkg when the nupkg is a duplicate.");//TODO: False to True for Success (and "expected to successfully push...")
+                    Assert.Contains(MESSAGE_RESPONSE_NO_SUCCESS, result.Errors); //nupkg duplicate TODO:  Contains from MESSAGE_RESPONSE_NO_SUCCESS to MESSAGE_EXISTING_PACKAGE
+                    Assert.DoesNotContain(MESSAGE_PACKAGE_PUSHED, result.Output); //snupkgFileName and snupkgFileName2 pushed
+                    Assert.DoesNotContain(genericFileNotFoundError, result.Errors);
+                    Assert.Contains(snupkgFileName, result.AllOutput); //first snupkg is attempted as push
+                    Assert.DoesNotContain(snupkgFileName2, result.AllOutput); //TODO: Should Contain second snupkg which is attempted when first duplicate is skipped.
+                }
+            }
+        }
+
+        /// <summary>
+        /// When pushing *.Snupkg, (no skip duplicate) a 409 Conflict is returned and halts the remaining symbols push.
+        /// TODO: bug fixes will come from https://github.com/NuGet/Home/issues/8148
+        /// </summary>
+        [Fact]
+        public void PushCommand_Server_Snupkg_WildcardFindsMatchingSnupkgs_Conflict()
```

Comment:
Isn't this an opposite scenario of what NuGet.org does?

Reply:
Nope. NuGet.org will return conflicts when a snupkg is being validated.

Reply:
Ok, makes sense. So NuGet.org might return either 409 or 201 depending on how much it's been since the last push.